intake/interface/source/defined_plots.py

```diff
@@ -128,15 +152,99 @@ def selected(self, selected):
         """When plot is selected set, make sure widget stays upto date"""
         self.select.value = selected
 
-    def callback(self, *events):
+    def plot_selected(self, *events):
         for event in events:
             if event.name == 'value':
                 self.pane.object = self._plot_object(event.new)
-            if event.name == 'options':
-                self.instructions.object = self.instructions_contents
+            self.custom.name = "Create" if str(self.select.value) == 'None' else "Edit"
+            self.edit_options.visible = self.custom.name == "Edit"
+
+    def name_changed(self, *events):
+        for event in events:
+            if event.name == "value_input":
+                # Empty name not allowed and name cannot already be in use
+                if len(event.new) == 0 or event.new in self.options:
+                    self.interact_save.disabled = True
+                else:
+                    self.interact_save.disabled = False
+
+    def cancel(self, _):
+        self.pane.object = self._plot_object(self.selected)
+        self.out[0] = self.pane
+        self.children[0].visible = True
+        self.children[1].visible = False
+        self._behavior_callback = None
+
+    def interact_action(self, _):
```

Comment:
This is the saving of the plot into the catalog? Then the catalog is written to the original location, since that's what default `add()` does.

So this is not a way to "save as...". Do you anticipate enabling that workflow in the GUI? How about taking a catalog which is not YAMLFile, and saving it as YAML (which you can usually do)? 

I think the full set of possible behaviours and how we can fit them in is the most critical thing in this PR. May well be worth asking a design colleague to take a look.

Reply:
The correct workflow feels like it should happen at the Sources level. If a source is a local YAML file, then plots become editable and it will save over the existing catalog. If a source is not editable, the plot editing options would be grayed out.

A new functionality needs to be added to a Source to make a copy and/or make a linked wrapper. But this is part of the bigger need for sharing catalogs, nesting catalogs, pointing to other catalogs, etc from the GUI. I think that can wait for a future PR as the scope is quite large.